src/bank.rs

```diff
@@ -819,12 +925,84 @@ mod tests {
     }
 
     #[test]
-    fn test_one_tx_two_out_atomic_fail() {
+    fn test_unlock_accounts() {
+        let mint = Mint::new(2);
+        let key1 = Keypair::new().pubkey();
+        let bank = Bank::new(&mint);
+        assert_eq!(bank.last_id(), mint.last_id());
+        {
+            let last_id = hash(bank.last_id().as_ref());
+            bank.register_entry_id(&last_id);
+            let tx = Transaction::system_move(&mint.keypair(), key1, 3, last_id, 0);
+            let res = bank.process_transactions(&vec![tx.clone()]);
+            assert!(res[0].is_err());
+            assert_eq!(
+                bank.get_signature(&tx.last_id, &tx.signature),
+                Some(Err(BankError::ResultWithNegativeTokens(0))),
+            );
+        }
+        {
+            let last_id = hash(bank.last_id().as_ref());
+            bank.register_entry_id(&last_id);
+            let tx = Transaction::system_move(&mint.keypair(), key1, 1, last_id, 0);
+            let res = bank.process_transactions(&vec![tx.clone()]);
+            assert!(res[0].is_ok());
```

Comment:
Let's get that `assert_eq!(x, Ok(()))` technique in all the new tests. That'll really make it easy to debug tests when a patch causes the function to return an error.